server/controllers/applications/getMyApplication.js

```diff
@@ -1,3 +1,44 @@
-module.exports = (req, res) => {
-  res.send('get my appplication');
+const yup = require('yup');
+
+const getMyApplications = require('../../database/queries/applications/getMyApplications');
+const getMyApplication = require('../../database/queries/applications/getMyApplication');
```

Comment:
@AymanAlqoqa Sometimes it's simpler for the client if you create two endpoints, instead of having one endpoint have two different behaviors like this (when the query param is present, it returns one application for that offer).

So an idea for the future - you could set up the endpoints like this:
`api/v1/:memberId/my-applications` - returns all applications for a member
`api/v1/:memberId/my-applications/:offerId` - returns one application for the member
That is more "restful" since the id (the memberId) that "owns" the other type (the application) appears before that type in the url.  When the owned type appears first, it's confusing to the client because they don't know if they should put the memberId, or the applicationId, after the "my-application/" in the url.  In other words, if someone else is writing the client, they wouldn't know if they should request:
`api/v1/my-applications/:memberId` or
`api/v/1my-applications/:offerId`.

Reply:
you are right :)
thanks